backend/src/middleware/notifications/notificationsMiddleware.spec.js

```diff
@@ -541,5 +573,148 @@ describe('notifications', () => {
         })
       })
     })
+
+    describe('given I file a report on a', () => {
+      let resourceId
+      const reasonCategory = 'discrimination_etc'
+      const reasonDescription = 'I am free to be gay !!!'
+      let reportFiler
+      let reportedUserOrAuthorData
+      const fileReportAction = async () => {
+        authenticatedUser = await reportFiler.toJson()
+        await mutate({
+          mutation: fileReportMutation,
+          variables: {
+            resourceId,
+            reasonCategory,
+            reasonDescription,
+          },
+        })
+        authenticatedUser = await notifiedUser.toJson()
+      }
+      const setExpectedNotificationOfReportedResource = reportedResource => {
+        return expect.objectContaining({
+          data: {
+            notifications: [
+              {
+                createdAt: expect.any(String),
+                read: false,
+                reason: 'filed_report_on_resource',
+                from: {
+                  __typename: 'Report',
+                  id: expect.any(String),
+                  filed: [
+                    {
+                      reasonCategory: 'discrimination_etc',
+                      reasonDescription: 'I am free to be gay !!!',
+                      reportedResource,
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        })
+      }
+
+      beforeEach(async () => {
+        reportFiler = notifiedUser
+        reportedUserOrAuthorData = {
+          id: 'reportedUser',
+          name: 'Mrs Badman',
+          slug: 'mrs-badman',
+          email: 'reported-user@example.org',
+          password: '1234',
+        }
+      })
+
+      describe('user', () => {
+        it('sends me a notification for filing a report on a user', async () => {
+          await neode.create('User', reportedUserOrAuthorData)
```

Comment:
Better use our factories here, e.g. `email` will be stored in the node, but we use a separate node for a long time already.

Reply:
So true. @roschaefer 
I refactored all the tests and removed neode completely from this test.